### ISSUE TITLE
Building fails when passing BETTERCAMERA=0

### ISSUE THREAD

**Issue by vanfanel** [djoslin0/sm64ex-coop#156]
Hi there

I just tried to build on Debian 11 Stable with the `BETTERCAMERA=0` option and it fails:
```

In file included from src/game/behavior_actions.c:142:
src/game/behaviors/door.inc.c: In function 'door_allow_walk_through':
src/game/behaviors/door.inc.c:29:30: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
   29 |         if (gCamera->mode != CAMERA_MODE_NEWCAM) { return FALSE; }
      |                              ^~~~~~~~~~~~~~~~~~
      |                              CAMERA_MODE_NONE
src/game/behaviors/door.inc.c:29:30: note: each undeclared identifier is reported only once for each function it appears in
Piping: sound/samples/instruments/0B.aiff -> build/us_pc/sound/samples/instruments/0B.table.inc.c
Extracting codebook: sound/samples/instruments/0C_conga_stick.aiff -> build/us_pc/sound/samples/instruments/0C_conga_stick.table                                                                                                                  
In file included from src/game/behavior_actions.c:142:
src/game/behaviors/door.inc.c: In function 'set_door_camera_event':
src/game/behaviors/door.inc.c:45:26: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
   45 |     if (gCamera->mode == CAMERA_MODE_NEWCAM) {
      |                          ^~~~~~~~~~~~~~~~~~
      |                          CAMERA_MODE_NONE

```

**Comment by Bloxxel64** [comment]
> Hi there
> 
> I just tried to build on Debian 11 Stable with the `BETTERCAMERA=0` option and it fails:
> 
> ```
> 
> In file included from src/game/behavior_actions.c:142:
> src/game/behaviors/door.inc.c: In function 'door_allow_walk_through':
> src/game/behaviors/door.inc.c:29:30: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
>    29 |         if (gCamera->mode != CAMERA_MODE_NEWCAM) { return FALSE; }
>       |                              ^~~~~~~~~~~~~~~~~~
>       |                              CAMERA_MODE_NONE
> src/game/behaviors/door.inc.c:29:30: note: each undeclared identifier is reported only once for each function it appears in
> Piping: sound/samples/instruments/0B.aiff -> build/us_pc/sound/samples/instruments/0B.table.inc.c
> Extracting codebook: sound/samples/instruments/0C_conga_stick.aiff -> build/us_pc/sound/samples/instruments/0C_conga_stick.table                                                                                                                  
> In file included from src/game/behavior_actions.c:142:
> src/game/behaviors/door.inc.c: In function 'set_door_camera_event':
> src/game/behaviors/door.inc.c:45:26: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
>    45 |     if (gCamera->mode == CAMERA_MODE_NEWCAM) {
>       |                          ^~~~~~~~~~~~~~~~~~
>       |                          CAMERA_MODE_NONE
> ```

What are you looking to add with bettercam?

**Comment by vanfanel** [comment]
> > Hi there
> > I just tried to build on Debian 11 Stable with the `BETTERCAMERA=0` option and it fails:
> > ```
> > 
> > In file included from src/game/behavior_actions.c:142:
> > src/game/behaviors/door.inc.c: In function 'door_allow_walk_through':
> > src/game/behaviors/door.inc.c:29:30: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
> >    29 |         if (gCamera->mode != CAMERA_MODE_NEWCAM) { return FALSE; }
> >       |                              ^~~~~~~~~~~~~~~~~~
> >       |                              CAMERA_MODE_NONE
> > src/game/behaviors/door.inc.c:29:30: note: each undeclared identifier is reported only once for each function it appears in
> > Piping: sound/samples/instruments/0B.aiff -> build/us_pc/sound/samples/instruments/0B.table.inc.c
> > Extracting codebook: sound/samples/instruments/0C_conga_stick.aiff -> build/us_pc/sound/samples/instruments/0C_conga_stick.table                                                                                                                  
> > In file included from src/game/behavior_actions.c:142:
> > src/game/behaviors/door.inc.c: In function 'set_door_camera_event':
> > src/game/behaviors/door.inc.c:45:26: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
> >    45 |     if (gCamera->mode == CAMERA_MODE_NEWCAM) {
> >       |                          ^~~~~~~~~~~~~~~~~~
> >       |                          CAMERA_MODE_NONE
> > ```
> 
> What are you looking to add with bettercam?

I am not trying to add anything, I am trying to disable a feature.

**Comment by Bloxxel64** [comment]
> > > Hi there
> > > I just tried to build on Debian 11 Stable with the `BETTERCAMERA=0` option and it fails:
> > > ```
> > > 
> > > In file included from src/game/behavior_actions.c:142:
> > > src/game/behaviors/door.inc.c: In function 'door_allow_walk_through':
> > > src/game/behaviors/door.inc.c:29:30: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
> > >    29 |         if (gCamera->mode != CAMERA_MODE_NEWCAM) { return FALSE; }
> > >       |                              ^~~~~~~~~~~~~~~~~~
> > >       |                              CAMERA_MODE_NONE
> > > src/game/behaviors/door.inc.c:29:30: note: each undeclared identifier is reported only once for each function it appears in
> > > Piping: sound/samples/instruments/0B.aiff -> build/us_pc/sound/samples/instruments/0B.table.inc.c
> > > Extracting codebook: sound/samples/instruments/0C_conga_stick.aiff -> build/us_pc/sound/samples/instruments/0C_conga_stick.table                                                                                                                  
> > > In file included from src/game/behavior_actions.c:142:
> > > src/game/behaviors/door.inc.c: In function 'set_door_camera_event':
> > > src/game/behaviors/door.inc.c:45:26: error: 'CAMERA_MODE_NEWCAM' undeclared (first use in this function); did you mean 'CAMERA_MODE_NONE'?
> > >    45 |     if (gCamera->mode == CAMERA_MODE_NEWCAM) {
> > >       |                          ^~~~~~~~~~~~~~~~~~
> > >       |                          CAMERA_MODE_NONE
> > > ```
> > 
> > 
> > What are you looking to add with bettercam?
> 
> I am not trying to add anything, I am trying to disable a feature.

which one?

**Comment by vanfanel** [comment]
@Bloxxel64 puppycam

**Comment by Bloxxel64** [comment]
> @Bloxxel64 puppycam

there is none. is default cam.

**Comment by vanfanel** [comment]
> > @Bloxxel64 puppycam
> 
> there is none. is default cam.

NO: there's a default cam, and the puppycam.
This conversation has no sense at all.

**Comment by EmeraldLoc** [comment]
Ok, so, coop does not need better cam, it has better cam without that flag, so the flag should actually be removed from the makefile, I don't know why it is not, but it should.

**Comment by djoslin0** [comment]
Yeah it should be removed as a compile flag. You can toggle the "bettercam" on and off using the in-game menu
